### PR TITLE
Promote core.gardener.cloud/v1alpha1 to v1beta1

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
With this PR, every resource except `ShootState` gets promoted to `v1beta1` from `core.gardener.cloud/v1alpha1`.
In order to finally remove `garden.sapcloud.io/v1beta1` we should have at least a `beta` alternative for our end-users/stakeholders.

**Special notes for your reviewer**:
* All components (controller-manager, scheduler) still work with the `v1alpha1` version but will be adapted in a future version to `v1beta1` (after extension controllers had some time to be able to work with both `v1alpha1` and `v1beta1`).
* Changes compared to `v1alpha1`:
  * The `.spec.seed` field in `core.gardener.cloud/v1alpha1.BackupBucket` has been renamed to `.spec.seedName` in `core.gardener.cloud/v1beta1.BackupBucket`.
  * The `.spec.seed` field in `core.gardener.cloud/v1alpha1.BackupEntry` has been renamed to `.spec.seedName` in `core.gardener.cloud/v1beta1.BackupEntry`.
  * The `.spec.blockCIDRs` field `core.gardener.cloud/v1alpha1.Seed` has been moved to `.spec.networks.blockCIDRs` in `core.gardener.cloud/v1beta1.Seed`.
  * The `.spec.addons.kubernetes-dashboard` field `core.gardener.cloud/v1alpha1.Shoot` has been renamed to `.spec.addons.kubernetesDashboard` in `core.gardener.cloud/v1beta1.Shoot`.
  * The `.spec.addons.nginx-ingress` field `core.gardener.cloud/v1alpha1.Shoot` has been renamed to `.spec.addons.nginxIngress` in `core.gardener.cloud/v1beta1.Shoot`.
  * The `.status.seed` field `core.gardener.cloud/v1alpha1.Shoot` has been renamed to `.status.seedName` in `core.gardener.cloud/v1beta1.Shoot`.
  * The `.status.lastError` field `core.gardener.cloud/v1alpha1.Shoot` does no longer exist in `core.gardener.cloud/v1beta1.Shoot` (in favour of `.status.lastErrors`).

**Release note**:

```noteworthy user
As preparation for the final removal of the already deprecated `garden.sapcloud.io/v1beta1` API group, all resources (except `ShootState`) available in `core.gardener.cloud/v1alpha1` are now promoted to `core.gardener.cloud/v1beta1` with the following changes:
* The `.spec.seed` field in `core.gardener.cloud/v1alpha1.BackupBucket` has been renamed to `.spec.seedName` in `core.gardener.cloud/v1beta1.BackupBucket`.
* The `.spec.seed` field in `core.gardener.cloud/v1alpha1.BackupEntry` has been renamed to `.spec.seedName` in `core.gardener.cloud/v1beta1.BackupEntry`.
* The `.spec.blockCIDRs` field `core.gardener.cloud/v1alpha1.Seed` has been moved to `.spec.networks.blockCIDRs` in `core.gardener.cloud/v1beta1.Seed`.
* The `.spec.addons.kubernetes-dashboard` field `core.gardener.cloud/v1alpha1.Shoot` has been renamed to `.spec.addons.kubernetesDashboard` in `core.gardener.cloud/v1beta1.Shoot`.
* The `.spec.addons.nginx-ingress` field `core.gardener.cloud/v1alpha1.Shoot` has been renamed to `.spec.addons.nginxIngress` in `core.gardener.cloud/v1beta1.Shoot`.
* The `.status.seed` field `core.gardener.cloud/v1alpha1.Shoot` has been renamed to `.status.seedName` in `core.gardener.cloud/v1beta1.Shoot`.
* The `.status.lastError` field `core.gardener.cloud/v1alpha1.Shoot` does no longer exist in `core.gardener.cloud/v1beta1.Shoot` (in favour of `.status.lastErrors`).
```
```action developer
Developers should re-run `./hack/dev-setup-register-gardener` in order to register the new `core.gardener.cloud/v1beta1` API group.
```
```noteworthy developer
All Gardener components are still working with `core.gardener.cloud/v1alpha1` but will soon switch to `core.gardener.cloud/v1beta1`. That means that the `extensions.gardener.cloud/v1alpha1.Cluster` resource will then contain the `core.gardener.cloud/v1beta1` resources only. Extension controllers should be prepared to be able to work with both the `v1alpha1` and the `v1beta1` version.
```

### CODE REVIEW

File: pkg/apis/core/install/install.go [hunk 1]
```diff
@@ -26,6 +27,8 @@ import (
 func Install(scheme *runtime.Scheme) {
 	utilruntime.Must(core.AddToScheme(scheme))
 	utilruntime.Must(v1alpha1.AddToScheme(scheme))
+	utilruntime.Must(v1beta1.AddToScheme(scheme))
 
 	utilruntime.Must(scheme.SetVersionPriority(v1alpha1.SchemeGroupVersion))
+	utilruntime.Must(scheme.SetVersionPriority(v1beta1.SchemeGroupVersion))
```
Comment (timuthy) [inline pkg/apis/core/install/install.go]: The doc string of `scheme.SetVersionPriority` says:
> the specified order overwrites any previously specified order for this group

```suggestion
	utilruntime.Must(scheme.SetVersionPriority(v1alpha1.SchemeGroupVersion, v1beta1.SchemeGroupVersion))
```

File: pkg/apis/garden/helper/helpers.go [hunk 2]
```diff
@@ -255,7 +255,7 @@ func TaintsHave(taints []garden.SeedTaint, key string) bool {
 
 // QuotaScope returns the scope of a quota scope reference.
 func QuotaScope(scopeRef corev1.ObjectReference) (string, error) {
-	if scopeRef.APIVersion == "core.gardener.cloud/v1alpha1" && scopeRef.Kind == "Project" {
+	if (scopeRef.APIVersion == "core.gardener.cloud/v1alpha1" || scopeRef.APIVersion == "core.gardener.cloud/v1beta1") && scopeRef.Kind == "Project" {
```
Comment (timuthy) [inline pkg/apis/garden/helper/helpers.go]:
```suggestion
        gvk := schema.FromAPIVersionAndKind(scopeRef.APIVersion, scopeRef.Kind)
	if gvk.Group == "core.gardener.cloud" && gvk.Kind == "Project" {
```

File: pkg/apis/garden/v1beta1/conversions.go [hunk 3]
```diff
@@ -407,7 +419,7 @@ func Convert_garden_QuotaSpec_To_v1beta1_QuotaSpec(in *garden.QuotaSpec, out *Qu
 		return err
 	}
 
-	if in.Scope.APIVersion == "core.gardener.cloud/v1alpha1" && in.Scope.Kind == "Project" {
+	if (in.Scope.APIVersion == "core.gardener.cloud/v1alpha1" || in.Scope.APIVersion == "core.gardener.cloud/v1beta1") && in.Scope.Kind == "Project" {
```
Comment (timuthy) [inline pkg/apis/garden/v1beta1/conversions.go]:
```suggestion
        gvk := schema.FromAPIVersionAndKind(in.Scope.APIVersion, in.Scope.Kind)
	if gvk.Group == "core.gardener.cloud" && gvk.Kind == "Project" {
```

File: charts/gardener/charts/application/templates/apiservice-v1beta1-core-gardener-cloud.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    heritage: "{{ .Release.Service }}"
+spec:
+  insecureSkipTLSVerify: {{ .Values.global.apiserver.insecureSkipTLSVerify }}
+  {{- if not .Values.global.apiserver.insecureSkipTLSVerify }}
+  caBundle: {{ required ".Values.global.apiserver.caBundle is required" (b64enc .Values.global.apiserver.caBundle) }}
+  {{- end }}
+  group: core.gardener.cloud
+  version: v1beta1
+  groupPriorityMinimum: 9
+  versionPriority: 9
```
Comment (timuthy) [inline charts/gardener/charts/application/templates/apiservice-v1beta1-core-gardener-cloud.yaml]: Don't you think it's already good to give it a higher prio than v1alpha1?

**Review by timuthy — CHANGES_REQUESTED** [review]

File: charts/gardener/charts/application/templates/apiservice-v1beta1-core-gardener-cloud.yaml (hunk 4 above)
Comment (rfranzke) [reply charts/gardener/charts/application/templates/apiservice-v1beta1-core-gardener-cloud.yaml]: You're right, thanks, done.

**Review by timuthy — DISMISSED** [review]
/lgtm - very nice!

File: docs/extensions/extension.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ But, some requirements don't match with those special resources or don't depend
 Gardener creates one `Extension` resource per registered extension type in `ControllerRegistration` per shoot. 
 
 ```yaml
-apiVersion: core.gardener.cloud/v1alpha1
+apiVersion: core.gardener.cloud/v1beta1
 kind: ControllerRegistration
 metadata:
   name: extension-example
```
Comment (ialidzhikov) [inline docs/extensions/extension.md:15]: I guess the apiVersion of the Shoot example in this file could be also updated to `core.gardener.cloud/v1beta1`.

File: docs/extensions/controllerregistration.md [hunk 6]
```diff
@@ -8,7 +8,7 @@ Similar to how `CloudProfile` or `Seed` resources get into the system, the Garde
 The specification mainly describes which of Gardener's extension CRDs are managed, for example:
 
 ```yaml
-apiVersion: core.gardener.cloud/v1alpha1
```
Comment (ialidzhikov) [inline docs/extensions/controllerregistration.md:11]: You could also update `docs/extensions/backupbucket.md` and `docs/extensions/backupentry.md`.

File: pkg/apis/core/v1beta1/constants/types_constants.go [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+	// SecretNameGardener is a constant for the name of a Kubernetes secret object that contains the client
+	// certificate and a kubeconfig for a shoot cluster. It is used by Gardener and can be used by extension
+	// controllers in order to communicate with the shoot's API server. The client certificate has administrator
+	// privileges.
+	SecretNameGardener = "gardener"
+
+	// DeploymentNameClusterAutoscaler is a constant for the name of a Kubernetes deployment object that contains
+	// the cluster-autoscaler pod.
+	DeploymentNameClusterAutoscaler = "cluster-autoscaler"
+	// DeploymentNameClusterAutoscaler is a constant for the name of a Kubernetes deployment object that contains
```
Comment (ialidzhikov) [inline pkg/apis/core/v1beta1/constants/types_constants.go]:
```suggestion
	// DeploymentNameDependencyWatchdog is a constant for the name of a Kubernetes deployment object that contains
```

File: pkg/apis/core/v1beta1/constants/types_constants.go [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+	// the dependency-watchdog pod.
+	DeploymentNameDependencyWatchdog = "dependency-watchdog"
+	// DeploymentNameClusterAutoscaler is a constant for the name of a Kubernetes deployment object that contains
+	// the kube-apiserver pod.
+	DeploymentNameKubeAPIServer = "kube-apiserver"
+	// DeploymentNameClusterAutoscaler is a constant for the name of a Kubernetes deployment object that contains
```
Comment (ialidzhikov) [inline pkg/apis/core/v1beta1/constants/types_constants.go]:
```suggestion
	// DeploymentNameKubeControllerManager is a constant for the name of a Kubernetes deployment object that contains
```

File: pkg/apis/core/v1beta1/constants/types_constants.go [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+	// PrometheusConfigMapAlertingRules is a constant for the Prometheus alerting rules tag in provider-specific monitoring configuration
+	PrometheusConfigMapAlertingRules = "alerting_rules"
+	// PrometheusConfigMapScrapeConfig is a constant for the Prometheus scrape config tag in provider-specific monitoring configuration
+	PrometheusConfigMapScrapeConfig = "scrape_config"
+	// GrafanaConfigMapUserDashboard is a constant for the Grafana user dashboard tag in provider-specific monitoring configuration
+	GrafanaConfigMapUserDashboard = "dashboard_users"
+	// GrafanaConfigMapOperatorDashboard is a constant for the Grafana operator dashboard tag in provider-specific monitoring configuration
+	GrafanaConfigMapOperatorDashboard = "dashboard_operators"
+
+	// ControllerRegistrationName is the key of a label on extension namespaces that indicates the controller registration name.
```
Comment (ialidzhikov) [inline pkg/apis/core/v1beta1/constants/types_constants.go:251]:
```suggestion
	// LabelControllerRegistrationName is the key of a label on extension namespaces that indicates the controller registration name
```

File: pkg/apis/core/v1beta1/types_cloudprofile.go [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +optional
+	VolumeTypes []VolumeType `json:"volumeTypes,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
+}
+
+// KubernetesSettings contains constraints regarding allowed values of the 'kubernetes' block in the Shoot specification.
+type KubernetesSettings struct {
+	// Versions is the list of allowed Kubernetes versions with optional expiration dates for Shoot clusters.
+	// +patchMergeKey=version
```
Comment (ialidzhikov) [inline pkg/apis/core/v1beta1/types_cloudprofile.go:89]: Hmm, should both patchMergeKey match - both `version` and `name` are specified?

File: pkg/apis/core/v1beta1/types_cloudprofile.go [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+	// +optional
+	Versions []ExpirableVersion `json:"versions,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
+}
+
+// MachineImage defines the name and multiple versions of the machine image in any environment.
+type MachineImage struct {
+	// Name is the name of the image.
+	Name string `json:"name"`
+	// Versions contains versions and expiration dates of the machine image
+	// +patchMergeKey=version
```
Comment (ialidzhikov) [inline pkg/apis/core/v1beta1/types_cloudprofile.go:100]: Same question about the patchMergeKey?

File: pkg/apis/garden/types.go [hunk 12]
```diff
@@ -643,6 +640,9 @@ type SeedNetworks struct {
 	Services string
 	// ShootDefaults contains the default networks CIDRs for shoots.
 	ShootDefaults *ShootNetworks
+	// BlockCIDRs is a list of network addresses tha should be blocked for shoot control plane components running
```
Comment (ialidzhikov) [inline pkg/apis/garden/types.go]:
```suggestion
	// BlockCIDRs is a list of network addresses that should be blocked for shoot control plane components running
```

**Review by ialidzhikov — COMMENTED** [review]
Minor nits.

File: docs/extensions/controllerregistration.md (hunk 6 above)
Comment (rfranzke) [reply docs/extensions/controllerregistration.md:11]: What exactly shall I update there?

File: pkg/apis/core/v1beta1/types_cloudprofile.go (hunk 10 above)
Comment (rfranzke) [reply pkg/apis/core/v1beta1/types_cloudprofile.go:89]: I don't know, please contact @mvladev. I just took what he contributed for `v1alpha1`. Let's do it outside of this PR.

File: pkg/apis/core/v1beta1/types_cloudprofile.go (hunk 11 above)
Comment (rfranzke) [reply pkg/apis/core/v1beta1/types_cloudprofile.go:100]: I don't know, please contact @mvladev. I just took what he contributed for `v1alpha1`. Let's do it outside of this PR.

**Review by ialidzhikov — DISMISSED** [review]

**Comment by rfranzke** [comment]
Thanks for your reviews, guys. Rebasing to resolve the conflicting files now and merging after CI passed.

**Review by ialidzhikov — APPROVED** [review]

**Pull request merged**